### PR TITLE
Parallel api changes

### PR DESCRIPTION
**Pull request by Oddant1**
Fixes and clean up the parallel API.

Withing in a `ParallelConfig` is now mandatory. 

You can no longer nest `ParallelConfigs.` 

`get_config` has been replaced by `get_config_from_file` and `get_config_from_dict`

### CODE REVIEW

File: qiime2/core/testing/tests/test_mapped_actions.py [hunk 1]
```diff
@@ -39,12 +30,6 @@ def run_non_pipeline(self, **inputs):
 
         return results
 
-    def run_pipeline(self, results, **inputs):
-        parsl_results = self.action.parallel(**inputs)._result()
-
-        for a, b in zip(results, parsl_results):
-            self.assertEqual(a.type, b.type)
-
```
Comment (Oddant1) [inline qiime2/core/testing/tests/test_mapped_actions.py:33]: This stuff was just not needed. This file never runs pipelines

**Comment by Oddant1** [comment]
This PR is blocking qiime2/dev-docs#61 and qiime2/q2cli#303

File: qiime2/sdk/parallel_config.py [hunk 2]
```diff
@@ -69,56 +67,59 @@ def setup_parallel(config_fp=None):
         config_fp = _get_vendored_config()
 
     if config_fp is not None:
-        config_dict = get_config(config_fp)
-        mapping = get_mapping(config_dict)
+        parallel_config, mapping = get_config(config_fp)
 
         # If config_dict is empty now, they gave a file that only contained a
```
Comment (lizgehret) [inline qiime2/sdk/parallel_config.py:72]:
```suggestion
        # If parallel_config is empty now, they gave a file that only contained a
```

**Review by lizgehret — COMMENTED** [review]
overall looks reasonable, couple of small things inline. once merge conflicts are resolved, this should be g2g

**Review by lizgehret — APPROVED** [review]
Only the most minor lint/copyright comment - everything else looks good, tests all pass and things play nicely locally.

**Pull request merged**